components/feature/session/src/main/java/mozilla/components/feature/session/SessionIntentProcessor.kt

```diff
@@ -0,0 +1,65 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package mozilla.components.feature.session
+
+import android.content.Intent
+import android.text.TextUtils
+
+typealias IntentHandler = (Intent) -> Boolean
+
+/**
+ * Processor for intents which should trigger session-related actions.
+ */
+class SessionIntentProcessor(
+    private val sessionUseCases: SessionUseCases,
+    useDefaultHandlers: Boolean = true
+) {
+    private val defaultActionViewHandler = { intent: Intent ->
+        val url = intent.dataString
+        if (TextUtils.isEmpty(url)) {
+            false
+        }
+
+        // TODO switch to loadUrlInNewTab, once available
+        sessionUseCases.loadUrl.invoke(url)
+        true
+    }
+
+    private val defaultHandlers: MutableMap<String, IntentHandler> by lazy {
+        mutableMapOf(Intent.ACTION_VIEW to defaultActionViewHandler)
+    }
```

Comment:
It may get a bit more complicated than that soon. "Custom Tabs" [1] are VIEW intents too. They just include some additional Intent "extras". I guess we could write a handler that handles both or delegates though. Anyhow, that's something we can always refactor later. :)

[1] https://developer.chrome.com/multidevice/android/customtabs